lists/arraylist/arraylist.go

```diff
@@ -99,14 +97,14 @@ func (list *List[T]) Contains(values ...T) bool {
 
 // Values returns all elements in the list.
 func (list *List[T]) Values() []T {
-	newElements := make([]T, list.size, list.size)
-	copy(newElements, list.elements[:list.size])
+	newElements := make([]T, len(list.elements))
```

Comment:
Can be replaced with `slices.Copy`

Reply:
There is no `slices.Copy()` in slices package. 😂

Reply:
Whoops, clone exists though https://pkg.go.dev/slices#Clone

Reply:
I created some benchmarks to test `Values()`. The `slices.Clone()` also improves performance when the array is large.

```diff
func (list *List[T]) Values() []T {
-	newElements := make([]T, len(list.elements))
-	copy(newElements, list.elements)
-	return newElements
+	return slices.Clone(list.elements)
}
```

```
goos: linux
goarch: amd64
pkg: github.com/emirpasic/gods/v2/lists/arraylist
cpu: Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
                        │   copy.txt   │           slices.clone.txt           │
                        │    sec/op    │    sec/op     vs base                │
ArrayListValues100-8      526.1n ± 15%   457.1n ± 17%        ~ (p=0.174 n=30)
ArrayListValues1000-8     4.606µ ±  7%   4.068µ ± 19%        ~ (p=0.096 n=30)
ArrayListValues10000-8    39.26µ ± 18%   34.02µ ± 18%        ~ (p=0.279 n=30)
ArrayListValues100000-8   375.3µ ± 17%   298.7µ ± 25%  -20.42% (p=0.004 n=30)
geomean                   13.75µ         11.72µ        -14.71%

                        │   copy.txt   │           slices.clone.txt            │
                        │     B/op     │     B/op      vs base                 │
ArrayListValues100-8        896.0 ± 0%     896.0 ± 0%       ~ (p=1.000 n=30) ¹
ArrayListValues1000-8     8.000Ki ± 0%   8.000Ki ± 0%       ~ (p=1.000 n=30) ¹
ArrayListValues10000-8    80.00Ki ± 0%   80.00Ki ± 0%       ~ (p=1.000 n=30) ¹
ArrayListValues100000-8   784.0Ki ± 0%   784.0Ki ± 0%  +0.00% (p=0.000 n=30)
geomean                   25.74Ki        25.74Ki       +0.00%
¹ all samples are equal

                        │  copy.txt  │          slices.clone.txt           │
                        │ allocs/op  │ allocs/op   vs base                 │
ArrayListValues100-8      1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=30) ¹
ArrayListValues1000-8     1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=30) ¹
ArrayListValues10000-8    1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=30) ¹
ArrayListValues100000-8   1.000 ± 0%   1.000 ± 0%       ~ (p=1.000 n=30) ¹
geomean                   1.000        1.000       +0.00%
¹ all samples are equal
```

```go
func benchmarkValues(b *testing.B, list *List[int]) {
	for i := 0; i < b.N; i++ {
		list.Values()
	}
}

func BenchmarkArrayListValues100000(b *testing.B) {
	b.StopTimer()
	size := 100000
	list := New[int]()
	for n := 0; n < size; n++ {
		list.Add(n)
	}
	b.StartTimer()
	benchmarkValues(b, list)
}
```

Reply:
@PapaCharlie 
Should I commit the benchmark I created in `arraylist_test.go`?